register-system.yaml

```diff
@@ -122,21 +139,38 @@
         state: present
       delegate_to: 127.0.0.1
 
-    - name: "Create network interfaces"
+    - name: "Create physical network interfaces"
       netbox.netbox.netbox_device_interface:
         netbox_url: "{{ netbox_url }}"
         netbox_token: "{{ netbox_token }}"
         data:
           device: "{{ inventory_hostname }}"
           name: "{{ item }}"
           mac_address: "{{ interface.macaddress|default('unknown') }}"
-          type: 10GBase-t (10GE)
+          type: >-
+            {{ speed_map[interface.speed|string] if interface.speed is defined else module_map[interface.module] | default('Other') }}
         state: present
       vars:
         interface: >-
           {{ hostvars[inventory_hostname]["ansible_{}".format(item.replace('-', '_'))] }}
       delegate_to: 127.0.0.1
-      loop: "{{ ansible_interfaces }}"
+      loop: "{{ physical_interfaces.stdout_lines }}"
+
+    - name: "Create virtual network interfaces that have an IP"
+      netbox.netbox.netbox_device_interface:
+        netbox_url: "{{ netbox_url }}"
+        netbox_token: "{{ netbox_token }}"
+        data:
+          device: "{{ inventory_hostname }}"
+          name: "{{ item }}"
+          mac_address: "{{ interface.macaddress|default('unknown') }}"
+          type: virtual
+        state: present
+      vars:
+        interface: >-
+          {{ hostvars[inventory_hostname]["ansible_{}".format(item.replace('-', '_'))] }}
+      delegate_to: 127.0.0.1
+      loop: "{{ansible_interfaces | difference(physical_interfaces.stdout_lines)}}"
```

Comment:
Clever use of the `difference` filter! I would have just put the check in the `when` condition, but this is better.